watertap/unit_models/electrocoagulation.py

```diff
@@ -0,0 +1,991 @@
+                    to_units=pyunits.volt,
+                )
+
+            @self.Expression(doc="Cathode cell potential via Nernst equation")
+            def cathode_cell_potential(b):
+                ccmh_dimensionless = pyunits.convert(
+                    b.cathode_conc_mol_hydroxide * pyunits.liter / pyunits.mol,
+                    to_units=pyunits.dimensionless,
+                )
+                ph2_dimensionless = pyunits.convert(
+                    b.partial_pressure_H2 * pyunits.atm**-1,
+                    to_units=pyunits.dimensionless,
+                )
+                return b.cathode_cell_potential_temp_adj - pyunits.convert(
+                    (
+                        (Constants.gas_constant * prop_out.temperature)
+                        / (Constants.faraday_constant * b.charge_transfer_number)
+                    ),
+                    to_units=pyunits.volt,
+                ) * log(ccmh_dimensionless**2 * ph2_dimensionless)
+
+            # Eq. 6 in Zhang et al. (2020); Eq. 12 in Dubrawski et al. (2014)
+            @self.Expression(doc="Anode activation overpotential")
+            def anode_overpotential(b):
+                return b.tafel_slope_anode * log(
+                    pyunits.convert(
+                        b.current_density / b.anodic_exchange_current_density,
+                        to_units=pyunits.dimensionless,
+                    )
+                )
+
+            # Eq. 7 in Zhang et al. (2020); Eq. 11 in Dubrawski et al. (2014)
+            @self.Expression(doc="Cathode activation overpotential")
+            def cathode_overpotential(b):
+                return -1 * (
+                    b.tafel_slope_cathode
+                    * log(
+                        pyunits.convert(
+                            b.current_density / b.cathodic_exchange_current_density,
+                            to_units=pyunits.dimensionless,
+                        )
+                    )
+                )
+
+            # See Eq. 3 in Zhang et al. (2020), Eq. 10 in Dubrawski et al. (2014)
+            # neglecting concentration/mass-transfer overpotential
+            @self.Constraint(doc="Overpotential calculation")
+            def eq_total_overpotential(b):
+                return b.overpotential == abs(
+                    b.cathode_cell_potential - b.anode_cell_potential
+                ) + b.anode_overpotential + abs(b.cathode_overpotential)
+
+        @self.Constraint(doc="Temperature change")
+        def eq_temperature_change(b):
+            return (
+                prop_out.temperature
+                == b.frac_increase_temperature * prop_in.temperature
+            )
+
+        @self.Constraint(doc="Water recovery")
+        def eq_water_recovery(b):
+            return (
+                prop_out.flow_mass_phase_comp["Liq", "H2O"]
+                == prop_in.flow_mass_phase_comp["Liq", "H2O"]
+                * b.recovery_frac_mass_water
```

Comment:
Just checked repo for `recovery_frac_mass_water` and I don't think this naming has been used anywhere. 

We do have the following for water in particular:

- `recovery_frac_mass_H2O` in ZO models (from TAP3)
- `recovery_mass_H2O` in ED_0D and ED_1D
- `recovery_mass_phase_comp` -- but this accounts for other components recoveries too
- `recovery_mass` in MD

Reply:
went with `recovery_mass_H2O`